### PR TITLE
ROC-8094: Auto enroll all claims into new features

### PR DESCRIPTION
**Pull request by AlokDatta**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-8094

### Change description
Introduced a new Launch Darkly feature toggle "autoEnrolIntoNewFeature"
added checks in newFeaturesConsentGuard.ts, FeaturesBuilder.ts and in create-claim-draft.ts
If the 'autoEnrolIntoNewFeature' is turned on, system will not display the consent to new feature page and any claims created will have new features by default.

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [x] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by AlokDatta** [comment]
> In Moacha.json, you have to enable autoEnrollIntoNewFeature to true, so that your test will run in Jenkins else your unit test will not run.

Hi Prathap, changes are made. Thanks.

File: src/main/features/claim/guards/newFeaturesConsentGuard.ts [hunk 1]
```diff
@@ -5,13 +5,18 @@ import { ClaimStoreClient } from 'claims/claimStoreClient'
...
+import { LaunchDarklyClient } from 'shared/clients/launchDarklyClient'
 
 const claimStoreClient = new ClaimStoreClient()
+const featureToggles: FeatureToggles = new FeatureToggles(new LaunchDarklyClient())
 
 export class NewFeaturesConsentGuard {
 
   static requestHandler (): express.RequestHandler {
     return GuardFactory.createAsync(async (req: express.Request, res: express.Response) => {
+      if (featureToggles.isAutoEnrollIntoNewFeatureEnabled()) {
```
Comment (prudhvi-v) [inline src/main/features/claim/guards/newFeaturesConsentGuard.ts:17]: Should this be an await ? isAutoEnrollIntoNewFeatreEnabled is an async function so.. where ever you call it it should be an async only

File: src/main/features/claim/helpers/featuresBuilder.ts [hunk 2]
```diff
@@ -19,7 +22,7 @@ export class FeaturesBuilder {
 
   async features (amount: number, user: User): Promise<string> {
     const roles: string[] = await this.claimStoreClient.retrieveUserRoles(user)
-    if (!roles.includes('cmc-new-features-consent-given')) {
+    if (!featureToggles.isAutoEnrollIntoNewFeatureEnabled() && !roles.includes('cmc-new-features-consent-given')) {
```
Comment (prudhvi-v) [inline src/main/features/claim/helpers/featuresBuilder.ts:25]: This should be an async function

File: src/main/features/testing-support/routes/create-claim-draft.ts [hunk 3]
```diff
@@ -34,7 +37,7 @@ export default express.Router()
 
       const roles: string[] = await claimStoreClient.retrieveUserRoles(user)
 
-      if (roles && !roles.some(role => role.includes('cmc-new-features-consent'))) {
+      if (!featureToggles.isAutoEnrollIntoNewFeatureEnabled() && roles && !roles.some(role => role.includes('cmc-new-features-consent'))) {
```
Comment (prudhvi-v) [inline src/main/features/testing-support/routes/create-claim-draft.ts:40]: Here as well. It should be an async

File: src/main/features/claim/guards/newFeaturesConsentGuard.ts (hunk 1 above)
Comment (AlokDatta) [reply src/main/features/claim/guards/newFeaturesConsentGuard.ts]: Acknowledged. Necessary changes are made as we have discussed.

File: src/main/features/claim/helpers/featuresBuilder.ts (hunk 2 above)
Comment (AlokDatta) [reply src/main/features/claim/helpers/featuresBuilder.ts]: Acknowledged. Necessary changes are made as we have discussed.

File: src/main/features/testing-support/routes/create-claim-draft.ts (hunk 3 above)
Comment (AlokDatta) [reply src/main/features/testing-support/routes/create-claim-draft.ts]: Acknowledged. Necessary changes are made as we have discussed.

**Pull request merged**